cirq/contrib/acquaintance/permutation.py

```diff
@@ -145,6 +145,14 @@ def __repr__(self):
     def _value_equality_values_(self):
         return (self.swap_gate,)
 
+    def _commutes_with_(self, other: Any, atol: Union[int, float] = 1e-8
+                       ) -> Union[bool, NotImplementedType]:
+        if (isinstance(other, ops.Gate) and
+                isinstance(other, ops.InterchangeableQubitsGate) and
+                protocols.num_qubits(other) == 2):
+            return True
```

Comment:
Is this correct? Swap gates only commute if they are on the same qubits or completely disjoint qubits. There's also the issue that the given swap gate may be asymmetric, but it could be considered a contract of the class.

Add an "assert_commutes_with_is_consistent_with_unitary" testing method to `cirq.testing`, which validates that when `_commutes_with_` returns false or true that it agrees with the results of the fallback computation on unitaries.

Reply:
At this point, gate commutation checks assume the gates act on exactly the same set of qubits.